### PR TITLE
Re #3827: added mutex to getADC(), because adcRead() is not reentrant

### PR DESCRIPTION
**Pull request by projectkk2glider**
This ~~might~~(https://github.com/opentx/opentx/issues/3827#issuecomment-252425167) also fix the #3827 original issue.

### CODE REVIEW

File: radio/src/opentx.cpp [hunk 1]
```diff
@@ -1138,8 +1138,6 @@ void checkTHR()
 
   while (1) {
 
-    SIMU_SLEEP(1);
-
     getADC();
```
Comment (bsongis) [inline radio/src/opentx.cpp]: In order to avoid the use of the mutex, we could check if the mixer is running, no?

File: radio/src/opentx.cpp [hunk 2]
```diff
@@ -1414,6 +1421,8 @@ uint16_t anaIn(uint8_t chan)
 #if defined(CPUARM)
 void getADC()
 {
+  static OS_MutexID getAdcMutex;
```
Comment (bsongis) [inline radio/src/opentx.cpp]: I thought there was an init method to be called, needs to be checked

**Comment by bsongis** [comment]
Good catch, this could explain the freeze / reboot on model load!

**Comment by projectkk2glider** [comment]
Do you like my solution without the mutex? 

I would also like you to try to reproduce the issue https://github.com/opentx/opentx/issues/3820#issuecomment-252413666

File: radio/src/opentx.h [hunk 3]
```diff
@@ -847,6 +847,11 @@ void checkAll();
 
 #if !defined(SIMU)
 void getADC();
+  #if defined(CPUARM)
+    #define getADC_ifMixerNotRunning()    do { if (s_pulses_paused) getADC(); } while(0)
```
Comment (bsongis) [inline radio/src/opentx.h]: I like this solution, yes
Instead I don't like this function name which mixes all existing styles around the world :)
May I propose GET_ADC_IF_MIXER_NOT_RUNNING()?

Comment (projectkk2glider) [inline radio/src/opentx.h]: I knew that :grin: I will change it.

**Comment by bsongis** [comment]
Thanks a lot for this problem hunting and fix!

**Pull request merged**